salt/transport/tcp.py

```diff
@@ -616,12 +816,20 @@ def getstream(self, **kwargs):
         stream = None
         while stream is None and (not self._closed and not self._closing):
             try:
-                stream = yield self._tcp_client.connect(
-                    ip_bracket(self.host, strip=True),
-                    self.port,
-                    ssl_options=self.opts.get("ssl"),
-                    **kwargs
-                )
+                if self.host and self.port:
+                    stream = await self._tcp_client.connect(
+                        ip_bracket(self.host, strip=True),
+                        self.port,
+                        ssl_options=self.opts.get("ssl"),
+                        **kwargs,
+                    )
+                else:
+                    sock_type = socket.AF_UNIX
+                    path = self.url.replace("ipc://", "")
```

Comment:
```suggestion
                    path = self.url.replace("ipc://", "")
```
I might have asked this before, but shouldn't this be `path = self.url.replace("ipc:/", "")`? Or do we want relative paths? or does the value actually looks like `ipc:///tmp/foo` ?

Reply:
This is in the MessageClient class and has been deprecated. We can safely ignore it.